driver-core/src/test/unit/com/mongodb/ConnectionStringSpecification.groovy

```diff
@@ -131,6 +124,35 @@ class ConnectionStringSpecification extends Specification {
         type << ['amp', 'semi', 'mixed']
     }
 
+    @Unroll
```

Comment:
Does this test describe all the substantive behavior changes in the ConnectionString class. Running this same test on master, 10 of them fail, and 5 succeed.  But are there other substantive behavioral changes that were made that are not tested here and only tested in the JSON-powered tests, and if so, what are those changes?